### DISCUSSION TITLE
Make a player send a chat message

### DISCUSSION THREAD

**Discussion by MrGeoTech** [Minestom/Minestom#1412] [Help] [answered]
Is there to make a player chat that will act the same as if the player sent the message? Sort of like `Player#chat()` for spigot. I have tried using [`Player#sendMessage(Identity, Component, MessageType)`](https://javadoc.minestom.net/net/minestom/server/entity/Player.html#sendMessage(net.kyori.adventure.identity.Identity,net.kyori.adventure.text.Component,net.kyori.adventure.audience.MessageType)) on every player but it still looks like it was sent from the server (no username etc). Am I just looking at the wrong thing?

**Comment by TheMode** [comment, marked as answer]
I'd not recommend such feature considering that chat signatures are coming.

One solution would be to broadcast a system message. Like we currently do in [Messenger](https://github.com/Minestom/Minestom/blob/18c46481f4b7915e18ee005c9e57f0c8073f3ba6/src/main/java/net/minestom/server/message/Messenger.java)

**Reply by MrGeoTech** [reply]
So I'm just going to have to make the message look like it was sent from the player even though it is a system message?

**Reply by TheMode** [reply]
We currently only send system messages, as chat signatures are not yet implemented on our side.

You can use `Player#addPacketToQueue` with a `ClientChatMessagePacket` to simulate a chat message. But be aware that this will break once we get signatures.

**Reply by MrGeoTech** [reply]
Hmm, ok. I guess I'll just do all the formatting in a util and then send the message to the players. Seems to be the only real way to do it with respect to chat signatures.
